core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -370,12 +370,10 @@ private[spark] class TaskSchedulerImpl(
       }
       if (!launchedAnyTask) {
         taskSet.abortIfCompletelyBlacklisted(hostToExecutors)
+      } else {
+        hasLaunchedTask = true
```

Comment:
I don't think this is a valid change to make, IIUC `hasLaunchedTask` is to identify whether there were any resources offered, it should not have anything to do with the locality mechanism.

Reply:
Hi, @jiangxb1987 Thanks for your comment.
From the usage of `hasLaunchedTask `  seems it is related to resources offer. But for the name itself, it is not correct. uh, maybe, trivial.